apis/python/src/tiledbsoma/factory.py

```diff
@@ -70,6 +77,10 @@ def _construct_member(
 
     if spec_name is None:
         raise SOMAError("internal error: spec_name was not found")
+    if encoding_version is None:
+        raise SOMAError("internal error: encoding_version not found")
+    if encoding_version != SOMA_ENCODING_VERSION:
+        raise SOMAError("Unsupported SOMA object encoding version")
```

Comment:
Should this be a ValueError?

Reply:
Maybe...I'm a bit unclear on our exception UX approach with code like this.  I'll change for now, as we definitely have some work to do in this area.